### PR TITLE
Feature/v0.9.15

### PR DESCRIPTION
**Pull request by svenseeberg**

### CODE REVIEW

File: bin/docserv [hunk 1]
```diff
@@ -232,41 +233,49 @@ class Deliverable:
...
+        cmd = shlex.split(command['cmd'])
+        logger.debug("Thread %i: %s" % (thread_id, cmd))
+        s = subprocess.Popen(cmd, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+        self.out, self.err = s.communicate()
+
+        if command['ret_val'] is not None and not command['ret_val'] == int(s.returncode):
+            logger.warning("Thread %i: Build failed! Unexpected return value %i for '%s'" % (
+                thread_id, s.returncode, command['cmd']))
+            logger.warning("Thread %i STDOUT: %s" % (thread_id, self.out.decode('utf-8')))
+            logger.warning("Thread %i STDERR: %s"% (thread_id, self.err.decode('utf-8')))
```
Comment (sknorr) [inline bin/docserv]: Do you need a space between `"%` here?

Comment (svenseeberg) [reply bin/docserv]: Not necessarily, but yes.

File: bin/docserv [hunk 2]
```diff
@@ -226,42 +260,86 @@ class Deliverable:
...
+                    break
+        if root_id:
+            bigfile = root_id
+            logger.debug("Found ROOTID for %s: %s" % ( self.id, root_id ))
         else:
-            cmd = command['cmd']
-        logger.debug("Thread %i: %s" % (thread_id, cmd))
-        cmd = shlex.split(cmd)
-        return cmd
+            bigfile = self.dc_file.replace('DC-', '')
```
Comment (svenseeberg) [inline bin/docserv]: @sknorr is this reasonable?

Comment (svenseeberg) [reply bin/docserv]: Not the question I wanted to ask.

File: bin/docserv [hunk 3]
```diff
@@ -226,42 +260,86 @@ class Deliverable:
...
+        command['cmd'] = command['cmd'].replace('__FILELIST__', ' '.join(self.path))
+        return command
+
+    def extract_root_id(self, command):
+        # extract root id from DC file and then title from bigfile
+        dc_path = os.path.join(self.parent.local_repo_build_dir, self.dc_file)
+        with open(dc_path) as f:
+            import re
+            for line in f.readlines():
+                m = re.search('ROOTID=\"(.+?)\"', line)
```
Comment (svenseeberg) [inline bin/docserv]: @sknorr is THIS reasonable?

Comment (sknorr) [reply bin/docserv]: Almost. I'd hunt for the line with `^\s*ROOTID\s*=` in the first step, and then remove `^\s*ROOTID\s*=\s*["']?` and `["']?\s*` in the end.

Comment (svenseeberg) [reply bin/docserv]: Thanks.

File: bin/docserv [hunk 4]
```diff
@@ -319,7 +319,8 @@ class Deliverable:
         with open(dc_path) as f:
             import re
             for line in f.readlines():
-                m = re.search('ROOTID=\"(.+?)\"', line)
+                #pylint: disable=W1401
+                m = re.search('^\s*ROOTID\s*=\s*[\"\']?(.+?)[\"\']?.*$', line.strip())
```
Comment (sknorr) [inline bin/docserv]: Oder so +1

**Comment by svenseeberg** [comment]
@sknorr 
```
/usr/bin/docserv-stitch --make-positive --valid-languages="en-us" /etc/docserv/config.d/ /tmp/docserv_stitch_4gwcl3i6
readlink: unrecognized option '--valid-languages=en-us'
Try 'readlink --help' for more information.
Directory --valid-languages=en-us does not exist.
```

**Pull request merged**